common/src/main/java/io/netty/util/internal/NativeLibraryLoader.java

```diff
@@ -371,4 +390,62 @@ private static void closeQuietly(Closeable c) {
     private NativeLibraryLoader() {
         // Utility
     }
+
+    private static final class NoexecVolumeDetector {
+
+        private static boolean canExecuteExecutable(File file)
+                throws ClassNotFoundException, NoSuchMethodException, InvocationTargetException,
+                       IllegalAccessException {
+            if (PlatformDependent.javaVersion() < 7) {
+                // Assume true because we
+                return true;
+            }
+
+            // If we can already execute, there is nothing to do.
+            if (file.canExecute()) {
+                return true;
+            }
+
+            // On volumes, with noexec set, even files with the executable POSIX permissions will fail to execute.
+            // So, check for executable permissions, then find out if the file can be executed.
+
+            // Everything is done reflectively to prevent the animal-sniffer plugin from complainin.
+            Class<?> filesClass = Class.forName("java.nio.file.Files");
```

Comment:
> it appears that it is fine to use Java7 classes provided that they are protected by a version check -- no?

No. If you need to use features not available in JDK6 you need a version check. Netty's minimum compatibility is currently at JDK6 so it must run against this JDK and we can't attempt to load any classes which won't exist in this context.

> My understanding of how Animal Sniffer works is that doing so would suppress the usages of these classes globally within the Netty codebase ... If you want, I could make a separate issue

I would prefer to have the code reference actual types and update the Animal sniffer configuration to provide more fine grained exceptions. For the purposes of this PR it is fine to add an exception using the current approach and we can update the animal sniffer exceptions in a followup PR.